lib/matplotlib/axes/_axes.py

```diff
@@ -5182,7 +5182,21 @@ def _pcolorargs(funcname, *args, **kw):
             return X, Y, C
 
         if len(args) == 3:
-            X, Y, C = [np.asanyarray(a) for a in args]
+            # Check x and y for bad data...
+            C = np.asanyarray(args[2])
+            X, Y = [cbook.safe_masked_invalid(a) for a in args[:2]]
+            if funcname == 'pcolormesh':
+                if np.ma.is_masked(X) or np.ma.is_masked(Y):
+                    raise TypeError(
```

Comment:
Or ValueError?

Reply:
Oops, sorry, I changed it, and then accidentally changed it back